--- nagios-base/html/includes/utils.inc.php	2013-08-30 19:46:14.000000000 +0200
+++ nagios-step/html/includes/utils.inc.php	2026-04-25 18:01:19.083424746 +0200
@@ -1,7 +1,13 @@
 <?php
 // HELPER UTILITIES
 
-require_once(dirname(__FILE__).'/../config.inc.php');
+// OMD: read site specific configuration file
+// OMD: extract sitename from script filename and find path to
+//      site specific configuration file
+$omd_parts = explode("/", $_SERVER["SCRIPT_FILENAME"]);
+$omd_sitename = $omd_parts[3];
+$omd_site_config_path = "/omd/sites/$omd_sitename/etc/nagios/config.inc.php";
+require_once($omd_site_config_path);
 
 function get_update_information(){
 	global $cfg;
