--- mod_fcgid-2.3.9.orig/modules/fcgid/fcgid_pm_unix.c	2026-04-25 17:40:39.192016248 +0200
+++ mod_fcgid-2.3.9/modules/fcgid/fcgid_pm_unix.c	2026-04-25 17:40:39.194852037 +0200
@@ -285,7 +285,8 @@
          * When running children as the configured user, the real uid
          * is switched.
          */
-        if (ap_unixd_config.suexec_enabled) {
+        //if (ap_unixd_config.suexec_enabled) {
+        if (0) {
             if (getuid() != 0) {
                 ap_log_error(APLOG_MARK, APLOG_EMERG, 0, main_server,
                              "mod_fcgid: current user is not root while suexec is enabled, exiting now");
