From 6080546f7bfffb1a294dc64a33c7f04dd40cbc41 Mon Sep 17 00:00:00 2001
From: Timotheus Bachinger <timotheus.bachinger@checkmk.com>
Date: Fri, 4 Apr 2025 14:54:08 +0200
Subject: [PATCH] Quick hack: avoid color codes in stdout

---
 mypy/private/mypy.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mypy/private/mypy.py b/mypy/private/mypy.py
index 2fadcba..1e7f6be 100644
--- a/mypy/private/mypy.py
+++ b/mypy/private/mypy.py
@@ -73,6 +73,7 @@ def run_mypy(
             "--explicit-package-bases",
             # speedup
             "--fast-module-lookup",
+            "--no-color-output",
         ]
         + srcs
     )
-- 
2.43.0

