diff --git a/src/google/protobuf/compiler/java/full/enum.cc b/src/google/protobuf/compiler/java/full/enum.cc
index 30db3b413..c0aad304b 100644
--- a/src/google/protobuf/compiler/java/full/enum.cc
+++ b/src/google/protobuf/compiler/java/full/enum.cc
@@ -28,6 +28,8 @@
 
 // Must be last.
 #include "google/protobuf/port_def.inc"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
 
 namespace google {
 namespace protobuf {
@@ -396,4 +398,5 @@ bool EnumNonLiteGenerator::CanUseEnumValues() {
 }  // namespace protobuf
 }  // namespace google
 
+#pragma GCC diagnostic pop
 #include "google/protobuf/port_undef.inc"
diff --git a/src/google/protobuf/compiler/java/full/message_builder.cc b/src/google/protobuf/compiler/java/full/message_builder.cc
index 53de5d324..6ed62596c 100644
--- a/src/google/protobuf/compiler/java/full/message_builder.cc
+++ b/src/google/protobuf/compiler/java/full/message_builder.cc
@@ -38,6 +38,8 @@
 
 // Must be last.
 #include "google/protobuf/port_def.inc"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
 
 namespace google {
 namespace protobuf {
@@ -816,4 +818,5 @@ void MessageBuilderGenerator::GenerateIsInitialized(io::Printer* printer) {
 }  // namespace protobuf
 }  // namespace google
 
+#pragma GCC diagnostic pop
 #include "google/protobuf/port_undef.inc"
diff --git a/src/google/protobuf/compiler/java/lite/enum.cc b/src/google/protobuf/compiler/java/lite/enum.cc
index 8e3c57cf9..b792cf0ee 100644
--- a/src/google/protobuf/compiler/java/lite/enum.cc
+++ b/src/google/protobuf/compiler/java/lite/enum.cc
@@ -25,6 +25,8 @@
 #include "google/protobuf/descriptor.pb.h"
 #include "google/protobuf/io/printer.h"
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
 namespace google {
 namespace protobuf {
 namespace compiler {
@@ -255,3 +257,4 @@ void EnumLiteGenerator::Generate(io::Printer* printer) {
 }  // namespace compiler
 }  // namespace protobuf
 }  // namespace google
+#pragma GCC diagnostic pop
