diff --git a/uv/private/create_venv.sh b/uv/private/create_venv.sh
index 3fcf296..33501fa 100644
--- a/uv/private/create_venv.sh
+++ b/uv/private/create_venv.sh
@@ -35,9 +35,9 @@ if [[ "$target" != /* ]]; then
 fi
 
 rm -rf "$target"
-"$UV" venv "$target" --python "$PYTHON"
+"$UV" venv --quiet "$target" --python "$PYTHON"
 source "$target/bin/activate"
-"$UV" pip install -r "$REQUIREMENTS_TXT" {{args}}
+"$UV" pip install --quiet -r "$REQUIREMENTS_TXT" {{args}}
 
 site_packages_extra_files=({{site_packages_extra_files}})
 if [ ! -z ${site_packages_extra_files+x} ]; then
