diff --git a/uv/private/venv.bzl b/uv/private/venv.bzl
--- a/uv/private/venv.bzl
+++ b/uv/private/venv.bzl
@@ -1,5 +1,6 @@
 "uv based venv generation"

+load("@rules_python//python:packaging.bzl", "PyWheelInfo")
 load(":interpreter_path.bzl", "python_interpreter_path")
 load(":transition_to_target.bzl", "transition_to_target")

@@ -18,5 +19,8 @@
             "{{destination_folder}}": ctx.attr.destination_folder,
             "{{site_packages_extra_files}}": " ".join(["'" + file.short_path + "'" for file in ctx.files.site_packages_extra_files]),
-            "{{args}}": " \\\n    ".join(ctx.attr.uv_args),
+            "{{args}}": " \\\n    ".join(
+                ctx.attr.uv_args +
+                [whl[PyWheelInfo].wheel.short_path for whl in ctx.attr.whls],
+            ),
         },
     )
@@ -24,8 +28,8 @@
 def _runfiles(ctx):
     py_toolchain = ctx.toolchains[_PY_TOOLCHAIN]
     runfiles = ctx.runfiles(
-        files = [ctx.file.requirements_txt] + ctx.files.site_packages_extra_files,
+        files = [ctx.file.requirements_txt] + ctx.files.site_packages_extra_files + ctx.files.whls,
         transitive_files = py_toolchain.py3_runtime.files,
     )
     runfiles = runfiles.merge(ctx.attr._uv[0].default_runfiles)
     return runfiles
@@ -46,6 +50,7 @@
         "_uv": attr.label(default = "@multitool//tools/uv", executable = True, cfg = transition_to_target),
         "template": attr.label(allow_single_file = True),
         "uv_args": attr.string_list(default = []),
+        "whls": attr.label_list(default = [], providers = [PyWheelInfo]),
     },
     toolchains = [_PY_TOOLCHAIN],
@@ -55,10 +60,18 @@
-def create_venv(name, requirements_txt = None, target_compatible_with = None, destination_folder = None, site_packages_extra_files = [], uv_args = []):
-    _venv(
+def create_venv(
+        name,
+        requirements_txt = None,
+        target_compatible_with = None,
+        destination_folder = None,
+        site_packages_extra_files = [],
+        uv_args = [],
+        whls = []):
+    _venv(
         name = name,
         destination_folder = destination_folder,
         site_packages_extra_files = site_packages_extra_files,
         requirements_txt = requirements_txt or "//:requirements.txt",
         target_compatible_with = target_compatible_with,
         uv_args = uv_args,
+        whls = whls,
         template = Label("//uv/private:create_venv.sh"),
     )
