commit 980d32e805aa84bf02f785aa575b4a86da90edd8
Author: Mathias Laurin <Mathias.Laurin+github.com@gmail.com>
Date:   Mon Feb 10 16:03:04 2025 +0100

    Sort sections of MYPYPATH
    
    makes debugging significantly easier

diff --git a/mypy/private/mypy.bzl b/mypy/private/mypy.bzl
index 105c9ae..c35187b 100644
--- a/mypy/private/mypy.bzl
+++ b/mypy/private/mypy.bzl
@@ -133,16 +133,14 @@ def _mypy_impl(target, ctx):
         # and as a way to skip iterating over depset contents to find generated
         # file roots?
 
-    unique_imports_dirs = imports_dirs.keys()
-    unique_generated_dirs = generated_dirs.keys()
     generated_imports_dirs = []
-    for generated_dir in unique_generated_dirs:
-        for import_ in unique_imports_dirs:
+    for generated_dir in generated_dirs.keys():
+        for import_ in imports_dirs.keys():
             generated_imports_dirs.append("{}/{}".format(generated_dir, import_))
 
     # types need to appear first in the mypy path since the module directories
     # are the same and mypy resolves the first ones, first.
-    mypy_path = ":".join(types + external_deps.keys() + unique_imports_dirs + unique_generated_dirs + generated_imports_dirs)
+    mypy_path = ":".join(sorted(types) + sorted(external_deps) + sorted(imports_dirs) + sorted(generated_dirs) + sorted(generated_imports_dirs))
 
     output_file = ctx.actions.declare_file(ctx.rule.attr.name + ".mypy_stdout")
 
