diff --git a/lint/clang_tidy.bzl b/lint/clang_tidy.bzl
index 31f4d15..dddbd23 100644
--- a/lint/clang_tidy.bzl
+++ b/lint/clang_tidy.bzl
@@ -262,6 +262,12 @@ def _get_args(ctx, compilation_context, srcs):
     args.extend([src.short_path for src in srcs])
     return args
 
+def _quote_arg(arg):
+    # Do not lose quotation marks when passing args to bash
+    if '\"' in arg and not arg.startswith("'") and not arg.endswith("'"):
+        return "'%s'" % arg
+    return arg
+
 def _get_compiler_args(ctx, compilation_context, srcs):
     # add args specified by the toolchain, on the command line and rule copts
     args = []
@@ -358,7 +364,7 @@ def clang_tidy_fix(ctx, compilation_context, executable, srcs, patch, stdout, ex
     """
     patch_cfg = ctx.actions.declare_file("_{}.patch_cfg".format(ctx.label.name))
     clang_tidy_args = _get_args(ctx, compilation_context, srcs)
-    compiler_args = _get_compiler_args(ctx, compilation_context, srcs)
+    compiler_args = [_quote_arg(arg) for arg in _get_compiler_args(ctx, compilation_context, srcs)]
 
     ctx.actions.write(
         output = patch_cfg,
