diff --git a/format/private/format.sh b/format/private/format.sh
index 4af96e2..a9bda75 100755
--- a/format/private/format.sh
+++ b/format/private/format.sh
@@ -145,6 +145,7 @@ function ls-files {
       'Less') patterns=('*.less') ;;
       'Markdown') patterns=('contents.lr' '*.md' '*.livemd' '*.markdown' '*.mdown' '*.mdwn' '*.mkd' '*.mkdn' '*.mkdown' '*.ronn' '*.scd' '*.workbook') ;;
       'Pkl') patterns=('*.pkl' 'PklProject') ;;
+      'PowerShell') patterns=('*.ps1' '*.psm1' '*.psd1') ;;
       'Protocol Buffer') patterns=('*.proto') ;;
       'Python') patterns=('.gclient' 'DEPS' 'SConscript' 'SConstruct' 'wscript' '*.py' '*.cgi' '*.fcgi' '*.gyp' '*.gypi' '*.lmi' '*.py3' '*.pyde' '*.pyi' '*.pyp' '*.pyt' '*.pyw' '*.rpy' '*.spec' '*.tac' '*.wsgi' '*.xpy') ;;
       # GitHub Linguist also groups .qbs with QML, but qmlformat only supports .qml files.
diff --git a/format/private/formatter_binary.bzl b/format/private/formatter_binary.bzl
index 1b0c64e..27892b8 100644
--- a/format/private/formatter_binary.bzl
+++ b/format/private/formatter_binary.bzl
@@ -34,6 +34,7 @@ TOOLS = {
     "XML": "prettier",
     "Gherkin": "prettier",
     "Groovy": "groovy-format",
+    "PowerShell": "ps-format",
     "F#": "fantomas",
     "C#": "csharpier",
     "Pkl": "pkl",
@@ -72,6 +73,7 @@ CHECK_FLAGS = {
     "scalafmt": "--test --respect-project-filters",
     "clang-format": "--style=file --fallback-style=none --dry-run -Werror",
     "yamlfmt": "-lint",
+    "ps-format": "--check",
     "rustfmt": "--check",
     "fantomas": "--check",
     "csharpier": "check",
@@ -107,6 +109,7 @@ FIX_FLAGS = {
     "scalafmt": "--respect-project-filters",
     "clang-format": "-style=file --fallback-style=none -i",
     "yamlfmt": "",
+    "ps-format": "",
     "rustfmt": "",
     "fantomas": "",
     "csharpier": "format",
