diff --git a/cc/private/toolchain/unix_cc_toolchain_config.bzl b/cc/private/toolchain/unix_cc_toolchain_config.bzl
index ee26a55..907637c 100644
--- a/cc/private/toolchain/unix_cc_toolchain_config.bzl
+++ b/cc/private/toolchain/unix_cc_toolchain_config.bzl
@@ -362,6 +362,23 @@ def _impl(ctx):
         enabled = False,
     )
 
+    static_libstdcpp_feature = feature(
+        name = "static_libstdcpp",
+        enabled = False,
+        flag_sets = [
+            flag_set(
+                actions = [
+                    ACTION_NAMES.cpp_link_executable,
+                    ACTION_NAMES.cpp_link_dynamic_library,
+                    ACTION_NAMES.lto_index_for_executable,
+                    ACTION_NAMES.lto_index_for_dynamic_library,
+                ],
+                flag_groups = [flag_group(flags = ["-static-libstdc++"])],
+            ),
+        ],
+        implies = ["static_link_cpp_runtimes"],
+    )
+
     default_compile_flags_feature = feature(
         name = "default_compile_flags",
         enabled = True,
@@ -1273,6 +1290,21 @@ def _impl(ctx):
             flag_set(
                 actions = all_link_actions + lto_index_actions,
                 flag_groups = [flag_group(flags = ctx.attr.link_libs)] if ctx.attr.link_libs else [],
+                with_features = [
+                    with_feature_set(not_features = ["static_link_cpp_runtimes"]),
+                ],
+            ),
+            flag_set(
+                actions = all_link_actions + lto_index_actions,
+                flag_groups = [flag_group(
+                    flags = [
+                        "-l:libstdc++.a" if "stdc++" in link_lib else link_lib
+                        for link_lib in ctx.attr.link_libs
+                    ],
+                )] if ctx.attr.link_libs else [],
+                with_features = [
+                    with_feature_set(features = ["static_link_cpp_runtimes"]),
+                ],
             ),
         ],
     )
@@ -1803,6 +1835,7 @@ def _impl(ctx):
             ubsan_feature,
             gcc_quoting_for_param_files_feature,
             static_link_cpp_runtimes_feature,
+            static_libstdcpp_feature,
         ] + (
             [
                 supports_start_end_lib_feature,
@@ -1853,6 +1886,7 @@ def _impl(ctx):
             ubsan_feature,
             gcc_quoting_for_param_files_feature,
             static_link_cpp_runtimes_feature,
+            static_libstdcpp_feature,
         ] + (
             [
                 supports_start_end_lib_feature,
