diff -ru pnp4nagios-0.6.24.orig/share/pnp/application/models/data.php pnp4nagios-0.6.24/share/pnp/application/models/data.php
--- pnp4nagios-0.6.24.orig/share/pnp/application/models/data.php	2014-07-30 12:32:23.000000000 +0200
+++ pnp4nagios-0.6.24/share/pnp/application/models/data.php	2014-11-20 11:57:02.317424854 +0100
@@ -655,31 +655,32 @@
         }
     }
 
-    public function getTimeRange($start=FALSE ,$end=FALSE ,$view="") {
-        if($end != FALSE){
-            // we are in a custom timerange
-            if(!is_numeric($end)){
-                $timestamp = strtotime($end);
-                if(!$timestamp){
-                    throw new Kohana_User_Exception('Wrong Format', "End -> $end");
-                }else{
-                    $end = $timestamp;
-                }
-            }
-        }else{
-            $end = time();
+    private function getTimePoint($spec) {
+        if (!$spec) {
+	    return time();
+        }
+        else if (is_numeric($spec)) {
+            return $spec; // already is a timestamp
         }
-        if($start != FALSE ){
-            // we are in a custom timerange
-            if(!is_numeric($start)){
-                $timestamp = strtotime($start);
-                if(!$timestamp){
-                    throw new Kohana_User_Exception('Wrong Format', "Start -> $start");
-                }else{
-                    $start = $timestamp;
-                }
+        else {
+	    $parts = explode('_', $spec);
+            if (count($parts) >= 6) {
+		$timestring = sprintf("%04d-%02d-%02d %02d:%02d:00",
+		   $parts[2], $parts[0], $parts[1], $parts[3], $parts[4], $parts[5]);
+		return strtotime($timestring);
+            }
+            else {
+		throw new Kohana_User_Exception('Wrong Format', "$spec");
             }
         }
+    }
+
+    public function getTimeRange($start=FALSE ,$end=FALSE ,$view="") {
+	if ($end != FALSE or $start != FALSE) {
+	    $end = $this->getTimePoint($end);
+            $start = $this->getTimePoint($start);
+        }
+
         if($start && $end){
             $timerange['title']   = Kohana::lang('common.timerange-selector-link');
             $timerange['start']   = $start;
