diff --git a/share/pnp/application/views/graph_content.php b/share/pnp/application/views/graph_content.php
--- a/share/pnp/application/views/graph_content.php
+++ b/share/pnp/application/views/graph_content.php
@@ -61,7 +61,7 @@ foreach($this->data->STRUCT as $key=>$value){
         echo "</td>\n";
 	echo "<td align=right>";
 	echo pnp::zoom_icon($value['MACRO']['HOSTNAME'],
-		$value['MACRO']['SERVICEDESC'],
+		urlencode($value['MACRO']['SERVICEDESC']),
 		$value['TIMERANGE']['start'],
 		$value['TIMERANGE']['end'],
 		$value['SOURCE'],
diff --git a/share/pnp/application/views/zoom.php b/share/pnp/application/views/zoom.php
--- a/share/pnp/application/views/zoom.php
+++ b/share/pnp/application/views/zoom.php
@@ -69,7 +69,7 @@ if(!empty($tpl)){
 }else{
     echo "<img src=\"image?source=$source"
 	."&host=$host"
-	."&srv=$srv"
+	."&srv=".urlencode($srv)
 	."&view=$view"
 	."&start=$start"
 	."&end=$end"
-- 
1.7.10.4

