--- pnp4nagios-0.6.24/share/pnp/application/models/auth.php.orig	2014-12-02 11:47:54.467333210 +0100
+++ pnp4nagios-0.6.24/share/pnp/application/models/auth.php	2014-12-02 11:48:49.700363116 +0100
@@ -96,6 +96,10 @@
             return TRUE;
         }
 
+        if (isset($_SERVER['SKIP_AUTHORIZATION']) && $_SERVER['SKIP_AUTHORIZATION'] == 1) {
+            return TRUE;
+        }
+
         if($host == "pnp-internal"){
             return TRUE;
         }
