From ef0b0e0bde7bb50b6b5d6144a53869bfc2416597 Mon Sep 17 00:00:00 2001
From: Sven Panne <sp@mathias-kettner.de>
Date: Fri, 9 Feb 2018 15:12:37 +0100
Subject: [PATCH] Use fallthrough attribute when available.

---
 util/util.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/util/util.h b/util/util.h
index a69d842..9f8216b 100644
--- a/util/util.h
+++ b/util/util.h
@@ -11,6 +11,16 @@ using std::string;
 
 #define arraysize(array) (int)(sizeof(array)/sizeof((array)[0]))
 
+#if !defined(FALLTHROUGH_INTENDED) && defined(__has_cpp_attribute)
+#if __has_cpp_attribute(fallthrough)
+#define FALLTHROUGH_INTENDED [[fallthrough]]
+#elif __has_cpp_attribute(gnu::fallthrough)
+#define FALLTHROUGH_INTENDED [[gnu::fallthrough]]
+#elif __has_cpp_attribute(clang::fallthrough)
+#define FALLTHROUGH_INTENDED [[clang::fallthrough]]
+#endif
+#endif
+
 #ifndef FALLTHROUGH_INTENDED
 #define FALLTHROUGH_INTENDED do { } while (0)
 #endif
-- 
2.14.1

