diff -Nur dest/cryptography-1.5.3.orig/src/_cffi_src/openssl/cms.py  dest/cryptography-1.5.3/src/_cffi_src/openssl/cms.py
--- dest/cryptography-1.5.3.orig/src/_cffi_src/openssl/cms.py	2016-11-06 04:05:05.000000000 +0100
+++ dest/cryptography-1.5.3/src/_cffi_src/openssl/cms.py	2016-11-16 10:49:05.136002243 +0100
@@ -68,6 +68,9 @@
 #if !defined(OPENSSL_NO_CMS)
 static const long Cryptography_HAS_CMS = 1;
 static const long Cryptography_HAS_CMS_BIO_FUNCTIONS = 1;
+#if !defined(CMS_DEBUG_DECRYPT)
+const long CMS_DEBUG_DECRYPT = 0;
+#endif
 #else
 static const long Cryptography_HAS_CMS = 0;
 static const long Cryptography_HAS_CMS_BIO_FUNCTIONS = 0;
