diff -ur dokuwiki/lib/tpl/dokuwiki/style.ini dokuwiki/lib/tpl/dokuwiki/style.ini.neu
--- a/lib/tpl/dokuwiki/style.ini     2015-02-24 21:01:05.000000000 +0100
+++ b/lib/tpl/dokuwiki/style.ini.neu 2015-07-08 10:16:50.770634674 +0200
@@ -79,7 +79,7 @@
 __missing__         = "#d30"            ; @ini_missing

 ; site and sidebar widths
-__site_width__      = "75em"            ; @ini_site_width
+__site_width__      = "90%"            ; @ini_site_width
 __sidebar_width__   = "16em"            ; @ini_sidebar_width
 ; cut off points for mobile devices
 __tablet_width__    = "800px"           ; @ini_tablet_width
