diff -Nur dokuwiki.orig/lib/exe/css.php dokuwiki/lib/exe/css.php
--- dokuwiki.orig/lib/exe/css.php	2016-07-03 18:08:51.000000000 +0200
+++ dokuwiki/lib/exe/css.php	2019-05-01 13:55:57.876291377 +0200
@@ -166,11 +166,11 @@
     global $conf;
 
     $less = new lessc();
-    $less->importDir[] = DOKU_INC;
+    $less->importDir = array(DOKU_INC);
     $less->setPreserveComments(!$conf['compress']);
 
     if (defined('DOKU_UNITTEST')){
-        $less->importDir[] = TMP_DIR;
+        $less->importDir = array(TMP_DIR);
     }
 
     try {
